dbt/tests/generic/test_column_length.sql

```diff
@@ -0,0 +1,46 @@
+-- Test that all columns in a list have the correct length.
+--
+-- Returns columns with the prefix `len_` representing the length of each
+-- columns for all rows where one of the columns has an incorrect length.
+--
+-- If additional columns should be returned along with the length columns,
+-- e.g. to add an identifiable key for the row, use the
+-- additional_select_columns argument.
+{% test column_length(
+  model, columns, length, additional_select_columns=[]
+) %}
+
+    {%- set columns_csv = additional_select_columns | join(', ') %}
+
+    {%- set length_columns = [] %}
+    {% for column in columns %}
+    {%- set length_columns = length_columns.append(
+        [column, 'len_' + column]
+    ) %}
+{% endfor %}
```

Comment:
The indentation for these `for` blocks is not quite right -- it seems like SQLFluff understands dbt Jinja templating but doesn't quite get `for` loops right. It also seems to be unsure about whether it wants logic to be indented inside the `{% test %}` and `{% macro %}` blocks. I'm going to investigate these issues in a follow-up PR.

Reply:
We can always turn off the fixer/linting for specific things like this, as you can enable/disable specific rules in the `.sqlfluff` config file. We could also switch using SQLFluff for the big views and CTAS and then use [sqlfmt](https://github.com/tconbeer/sqlfmt) for the jinja-heavy stuff.